container/core.py

```diff
@@ -169,27 +165,29 @@ def hostcmd_build(base_path, project_name, engine_name, var_file=None,
 def hostcmd_deploy(base_path, project_name, engine_name, var_file=None,
                    cache=True, **kwargs):
     assert_initialized(base_path)
-    logger.debug('Got extra args to `deploy` command', arguments=kwargs)
-    config = get_config(base_path, var_file=var_file, engine_name=engine_name)
+    config = get_config(base_path, var_file=var_file, engine_name=engine_name, project_name=project_name)
     local_images = kwargs.get('local_images')
     output_path = kwargs.pop('deployment_output_path', None) or config.deployment_path
 
     engine_obj = load_engine(['LOGIN', 'PUSH', 'DEPLOY'],
-                             engine_name, project_name or os.path.basename(base_path),
+                             engine_name, project_name,
                              config['services'], **kwargs)
 
     params = {
-        'deployment_output_path': os.path.normpath(os.path.expanduser(output_path)),
+        'deployment_output_path': os.path.normpath(os.path.abspath(os.path.expanduser(output_path))),
         'host_user_uid': os.getuid(),
         'host_user_gid': os.getgid(),
     }
     if config.get('settings', {}).get('k8s_auth'):
         params['k8s_auth'] = config['settings']['k8s_auth']
+    if config.get('settings', {}).get('k8s_namespace'):
```

Comment:
Reiterated comment about trying best we can to keep engine-specific snowflakes in the engine specification.